src/Jackett.Common/Models/IndexerConfig/ConfigurationDataBasicLoginOrApiKey.cs

```diff
@@ -0,0 +1,25 @@
+namespace Jackett.Common.Models.IndexerConfig
+{
+    public class ConfigurationDataBasicLoginOrAPIKey : ConfigurationData
+    {
+        public DisplayItem Instructions { get; private set; }
+        public bool UseAPIKey { get; private set; }
+
+        public ConfigurationDataBasicLoginOrAPIKey(string instructionMessageOptional = null, bool useApiKey = false)
+        {
+            UseAPIKey = useApiKey;
+            if (useApiKey)
+                AddDynamic("apikey", new StringItem { Name = "APIKey " });
```

Comment:
I think you should move here the code from `GazelleTracker` constructor to include the 2fa and usetokens fields in this class.